consensus/src/marshal/ingress/mailbox.rs

```diff
@@ -101,25 +118,37 @@ impl<V: Variant, B: Block> Mailbox<V, B> {
     }
 
     /// A request to retrieve the information about the highest finalized block.
-    pub async fn get_latest(&mut self) -> oneshot::Receiver<Option<(u64, B::Commitment)>> {
+    pub async fn get_info(
+        &mut self,
+        identifier: impl Into<Identifier<B::Commitment>>,
+    ) -> Option<(u64, B::Commitment)> {
         let (tx, rx) = oneshot::channel();
         if self
             .sender
-            .send(Message::GetLatest { response: tx })
+            .send(Message::GetInfo {
+                identifier: identifier.into(),
+                response: tx,
+            })
             .await
             .is_err()
         {
-            error!("failed to send get latest message to actor: receiver dropped");
+            error!("failed to send get info message to actor: receiver dropped");
+        }
+        match rx.await {
+            Ok(result) => result,
+            Err(_) => {
+                error!("failed to get info: receiver dropped");
+                None
+            }
         }
```

Comment:
worth returning `Result<Option<(...)>>`? More explicitly shows that something is wrong like the actor is down (or went down after enqueuing the request), but can be more cumbersome to work with than a pure `Option`